pkg/gatewayconfigurationserver/gatewayconfigurationserver.go

```diff
@@ -47,6 +49,56 @@ type GatewayConfigurationServer struct {
 	config *Config
 }
 
+func (gcs *GatewayConfigurationServer) makeGatewayConfigHandler(f func(context.Context, echo.Context, *ttnpb.Gateway) error) func(echo.Context) error {
```

Comment:
Nit: Maybe you can drop the word `Gateway` in the function names to make them shorter. Within the context of this service, it's clear that it's gateway config.